src/main/java/de/zalando/aruha/nakadi/enrichment/MetadataEnrichmentStrategy.java

```diff
@@ -0,0 +1,39 @@
+package de.zalando.aruha.nakadi.enrichment;
+
+import de.zalando.aruha.nakadi.domain.EventType;
+import de.zalando.aruha.nakadi.exceptions.EnrichmentException;
+import de.zalando.aruha.nakadi.util.FlowIdUtils;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public class MetadataEnrichmentStrategy implements EnrichmentStrategy {
+    @Override
+    public JSONObject enrich(final JSONObject event, final EventType eventType) throws EnrichmentException {
+        try {
+            final JSONObject metadata = event.getJSONObject("metadata");
+
+            setReceivedAt(metadata);
+            setEventTypeName(metadata, eventType);
+            setFlowId(metadata);
+        } catch (final JSONException e) {
+            throw new EnrichmentException("enrichment error", e);
+        }
+
+        return event;
+    }
+
+    private void setFlowId(final JSONObject metadata) {
+        metadata.put("flow_id", FlowIdUtils.peek());
```

Comment:
I didn't really look deep into it, but how do we ensure that we peek a right flowId here? At the end it's a static stack, right? Or is it somehow separated for each thread?